beacon/cached_provider_test.go

```diff
@@ -0,0 +1,90 @@
+package beacon
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+type mockBeaconService struct {
+	beacons map[types.EpochID]types.Beacon
+	calls   int
+}
+
+func newMockBeaconService() *mockBeaconService {
+	return &mockBeaconService{
+		beacons: make(map[types.EpochID]types.Beacon),
+	}
+}
+
+func (m *mockBeaconService) Beacon(_ context.Context, epoch types.EpochID) (types.Beacon, error) {
+	m.calls++
+	if b, ok := m.beacons[epoch]; ok {
+		return b, nil
+	}
+	return types.Beacon{}, errors.New("beacon not found")
+}
```

Comment:
What's the reason to not use `gomock` here?

Reply:
No particular reason. I will fix it to use a generated mock.